components/datetime/src/mock/mod.rs

```diff
@@ -87,13 +84,15 @@ pub fn parse_gregorian_from_str(input: &str) -> Result<DateTime<Gregorian>, Date
 /// ```
 pub fn parse_zoned_gregorian_from_str(
     input: &str,
-) -> Result<(DateTime<Gregorian>, MockTimeZone), DateTimeError> {
+) -> Result<(DateTime<Gregorian>, TimeZone), DateTimeError> {
     let datetime = parse_gregorian_from_str(input)?;
     let time_zone = match input
         .rfind(|c| c == '+' || /* ASCII */ c == '-' || /* U+2212 */ c == '−' || c == 'Z')
     {
         #[allow(clippy::indexing_slicing)] // TODO(#1668) Clippy exceptions need docs or fixing.
-        Some(index) => FromStr::from_str(&input[index..])?,
+        Some(index) => {
+            FromStr::from_str(&input[index..]).map_err(|_| DateTimeError::InvalidTimeZoneOffset)?
```

Comment:
Suggestion: there should be a variant `DateTimeError::TimeZone` that wraps any `TimeZoneError`, such that you end up with something like `DateTimeError::TimeZone(TimeZoneError::InvalidOffset)`

Reply:
Note that this would require a dependency from icu_calendar to icu_timezone

Are we okay with that?

Reply:
Why? This is in the `icu_datetime` crate

Reply:
I could also add it as a DTF error

But note that this is _only_ found in the mock APIs. We could also return Either or a different error

Reply:
Leaning towards Either since the mock APIs are test-only